app/meteor-accounts-saml/server/saml_server.js

```diff
@@ -186,6 +194,7 @@ Accounts.registerLoginHandler(function(loginRequest) {
 			if (username) {
 				newUser.username = username;
 			}
+			newUser.name = newUser.name || _guessNameFromUsername(newUser.username); // Make sure every user has a name as well
```

Comment:
newUser.username may still be undefined here in some setups, causing an error on the _guessNameFromUsername method. Please check if there's any value on it before calling the method.

Reply:
@Hudell I did not think this is possible, but the `if (username)` suggests it is 😬
Done.

Reply:
Yup, there are 3 possible scenarios:
 - SAML response includes an username;
 - Rocket.Chat generates a new username from other data (if configured to do so);
 - Neither of those.

Reply:
Ah, I didn’t remember the option to generate a username can be disabled. Thanks anyway!